lightning/src/ln/channel.rs

```diff
@@ -2032,6 +2061,20 @@ impl<Signer: Sign> Channel<Signer> {
 			}
 		}
 
+		if msg.amount_msat / 1000 < (self.feerate_per_kw as u64 * HTLC_TIMEOUT_TX_WEIGHT / 1000) + self.counterparty_dust_limit_satoshis {
```

Comment:
Did we agree to a multiplier here? Ie instead of checking with the *current* feerate, check with the current feerate times 2/4/whatever to prevent feerate increases from blowing up our dust? I think we should set the multiplier pretty high so that we can remove the counterparty feerate check entirely and only ever check the dust threshold.

Reply:
I think we agree in principle, but not on the heuristics to pick up an adequate feerate. In fact, if you want to offer us feerate bump margin w.r.t to our dust balance and that way authorize any feerate in `check_remote_fee` as long as it doesn't overflow `max_balance_dust_htlc_msat` we need to scale up the _accounted_ dust, not

Reply:
Hmm, the reason I like a multiplier is that, without any other information, if the feerate is high, it seems likely the feerate may increase significantly. If the feerate is low, probably it will only increase by 10sat/vb or so.

We could do something nonlinear, but I'm not a fan of a straight constant. Maybe we could always increase by max(10 sat/vb, 0.25x current feerate)?

Reply:
Okay so implemented the evaluation against a _preventive_ high-feerate, note tests coverage has been modified in consequence to avoid the newly inflated check of counterparty dust limit satoshi blocking failure against the holder limit (508_000 -> 2_300_000)

```
                let pending_remote_value_msat =
@@ -3482,6 +3496,10 @@ impl<Signer: Sign> Channel<Signer> {
                self.feerate_per_kw
        }
 
+       pub fn get_dust_buffer_feerate(&self) -> u32 {
+               cmp::max(2530, self.feerate_per_kw * 1250 / 1000)
+       }
+
```